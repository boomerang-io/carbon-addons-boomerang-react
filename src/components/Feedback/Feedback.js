import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { settings } from 'carbon-components';
import { Button } from 'carbon-components-react';
import {
  ModalHeader,
  ModalBody,
  ModalFooter,
} from 'carbon-components-react/lib/components/ComposedModal';

import HeaderMenuItem from '../HeaderMenuItem';

const { prefix } = settings;

class Feedback extends Component {
  render() {
    return (
      <HeaderMenuItem
        preventCloseOnClickOutside
        text="Submit an Idea"
        iconName="idea"
        className={`${prefix}--bmrg-feedback-container`}
      >
        {({ closeModal }) => {
          return (
            <>
              <ModalHeader title="Submit an Idea" closeModal={closeModal} />
              <ModalBody>
                <div className={`${prefix}--bmrg-feedback`}>
                  <p>
                    Share your thoughts and ideas on what we can do to improve the{' '}
                    {this.props.platformName} platform and our onboarding process.
                  </p>
                  <p>
                    To submit your idea, visit our portal at
                    {' ' /* We need to force a space before the link tag */}
                    <a
                      href={this.props.sendIdeasUrl}
                      target="_blank"
                      rel="noopener noreferrer"
                      aria-describedby="new-window-aria-desc-0"
                    >
                      {this.props.sendIdeasUrl}
                    </a>
                  </p>
                  <p>
                    For now, you must have an IBM email address and you will need to register with a
                    password during your first visit.
                  </p>
                  <p>
                    You will have the opportunity to see other public ideas, vote on them and track
                    the status of your idea.
                  </p>
                  <p>We look forward to your feedback!</p>
                </div>
              </ModalBody>
              <ModalFooter>
                <Button kind="primary" onClick={closeModal}>
                  OK
                </Button>
              </ModalFooter>
            </>
          );
        }}
      </HeaderMenuItem>
    );
  }
}

Feedback.propTypes = {
  sendIdeasUrl: PropTypes.string,
};

export default Feedback;
