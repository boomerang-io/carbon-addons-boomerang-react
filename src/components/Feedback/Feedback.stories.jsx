import Component from "./Feedback";

export default {
  title: "Platform/Feedback",
  component: Component,
  parameters: {
    docs: {
      description: {
        component: "Header menu item that displays modal with details on providing feedback to the platform",
      },
    },
  },
};

const Template = (args) => <Component {...args} />;

export const Feedback = Template.bind({});
Feedback.args = {
  platformName: "Boomerang",
  sendIdeasUrl: "https://ideas.ibm.com",
  platformOrganization: "IBM",
};