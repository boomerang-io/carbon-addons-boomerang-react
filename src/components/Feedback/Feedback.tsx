import React from "react";
import { prefix } from "../../internal/settings";
import { Button, ComposedModal, ModalHeader, ModalBody, ModalFooter } from "@carbon/react";
import HeaderMenuItem from "../Header/HeaderMenuItem";
import { Idea } from "@carbon/react/icons";

type Props = {
  closeModal: () => void;
  isOpen: boolean;
  platformName?: string;
  platformOrganization?: string;
  sendIdeasUrl?: string;
};

function Feedback(props: Props) {
  return (
    <ComposedModal
      aria-label="Feedback"
      open={props.isOpen}
      className={`${prefix}--bmrg-feedback-container ${prefix}--bmrg-header-modal`}
      onClose={props.closeModal}
      onKeyDown={(e: any) => e.stopPropagation()}
    >
      <ModalHeader title="Submit an Idea" closeModal={props.closeModal} />
      <ModalBody>
        <div className={`${prefix}--bmrg-feedback`}>
          <p>
            Share your thoughts and ideas on what we can do to improve the {props.platformName} platform and our
            onboarding process.
          </p>
          <p>
            To submit your idea, visit our portal at
            {" " /* We need to force a space before the link tag */}
            <a
              aria-describedby="new-window-aria-desc-0"
              href={props.sendIdeasUrl}
              target="_blank"
              rel="noopener noreferrer"
            >
              {props.sendIdeasUrl}
            </a>
            .
          </p>
          <p>
            For now, you must have an {props.platformOrganization} email address and you will need to register with a
            password during your first visit.
          </p>
          <p>
            You will have the opportunity to see other public ideas, vote on them and track the status of your idea.
          </p>
          <p>We look forward to your feedback!</p>
        </div>
      </ModalBody>
      <ModalFooter>
        <Button data-modal-primary-focus kind="primary" onClick={props.closeModal}>
          OK
        </Button>
      </ModalFooter>
    </ComposedModal>
  );
}

function FeedbackMenuItem(props: Omit<Props, "isOpen" | "closeModal">) {
  const menuItemRef = React.useRef<HTMLLinkElement>(null);
  const [isOpen, setIsOpen] = React.useState(false);

  const handleClose = () => {
    setIsOpen(false);
    setTimeout(() => {
      menuItemRef.current?.focus();
    }, 0);
  };

  return (
    <>
      <HeaderMenuItem
        icon={<Idea />}
        onClick={() => setIsOpen(!isOpen)}
        ref={menuItemRef}
        text="Submit an idea"
        type="button"
      />
      <Feedback isOpen={isOpen} closeModal={handleClose} {...props} />
    </>
  );
}

export { FeedbackMenuItem };

export default Feedback;
