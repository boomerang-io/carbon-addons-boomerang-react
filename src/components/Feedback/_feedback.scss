@include exports('addons-bmrg-feedback') {
  .#{$prefix}--bmrg-feedback-container {
    .#{$prefix}--modal-container {
      height: rem(448px);
    }

    .#{$prefix}--bmrg-feedback {
      color: $bmrg-brand-white;
      height: rem(100px);
      margin-top: rem(16px);
      width: 100%;

      a {
        color: $bmrg-header-modal-link-color;
      }

      p {
        margin-bottom: 1rem;
      }

      &__error {
        color: #fb4b53;
        font-size: 0.875rem;
        font-weight: 400;
        line-height: 1.125rem;
        margin-top: rem(4px);
        position: absolute;
      }

      .#{$prefix}--label {
        color: $bmrg-brand-white;
        font-size: rem(14px);
        font-weight: 500;
        line-height: rem(18px);
      }
    }
  }
}
