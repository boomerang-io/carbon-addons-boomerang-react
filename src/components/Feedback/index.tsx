export { default, FeedbackMenuItem } from "./Feedback";
