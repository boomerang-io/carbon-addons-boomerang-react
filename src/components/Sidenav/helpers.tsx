import React from "react";

export const header = () => <div style={{ fontSize: "5rem", color: "#047cc0" }}>title</div>;
export const SidenavOutstandingTeams = ({ outstandingTeamRequests }: any) => {
  const numberOfOutstandingTeamRequests = outstandingTeamRequests.length;

  if (numberOfOutstandingTeamRequests === 0) {
    return null;
  }

  return (
    <>
      <div
        style={{
          height: "5rem",
          color: "black",
          textAlign: "center",
          marginTop: "1rem",
        }}
        data-tip
        data-for="outstandingTeamRequests"
      >{`${numberOfOutstandingTeamRequests} ${
        numberOfOutstandingTeamRequests === 1 ? "REQUEST" : "REQUESTS"
      } PENDING`}</div>
    </>
  );
};

export const outstandingTeamRequests = [
  {
    id: "59e5720c0aa79f128e639448",
    groupName: "Pending Test Team",
  },
];

export const navItems = [
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: true,
    text: "Demo team 1",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home",
    exact: true,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 21",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home1",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 223",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home2",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 243",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home3",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 254657",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home4",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2345345",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home5",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2334533",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home6",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 24345345345345",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home7",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2334535345345345",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home8",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 256657657653424786783323",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home9",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home10",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 24557423432",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home11",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 12",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home12",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2564",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home13",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 234",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home14",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 288676443",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home15",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2222",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home1",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 25555",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home16",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2666",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home17",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2899",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home18",
    exact: false,
  },
];
