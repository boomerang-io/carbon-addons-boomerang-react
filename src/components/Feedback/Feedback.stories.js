import React from 'react';
import { storiesOf } from '@storybook/react';
import Feedback from './Feedback';

storiesOf('Feedback', module).add('default', () => {
  return (
    <>
      <Feedback
        platformName="IBM Boomerang"
        sendIdeasUrl="https://ideas.ibm.com"
        platformOrganization="IBM"
      />
    </>
  );
});
