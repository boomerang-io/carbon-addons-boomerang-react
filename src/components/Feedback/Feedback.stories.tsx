import React from "react";
import Feedback from "./Feedback";
import { headerModalProps } from "../../internal/helpers";

export default {
  title: "Platform/Feedback",
  component: Feedback,
  parameters: {
    docs: {
      description: {
        component: "Header modal with details on providing feedback to the platform",
      },
    },
  },
};

export const Default = (args) => {
  return <Feedback {...args} />;
};

Default.args = {
  platformName: "Boomerang",
  sendIdeasUrl: "https://ideas.ibm.com",
  platformOrganization: "IBM",
  ...headerModalProps,
};
