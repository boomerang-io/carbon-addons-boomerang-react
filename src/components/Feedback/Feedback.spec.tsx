import React from "react";
import { expect, test } from "vitest";
import { render, fireEvent, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import Feedback from "./Feedback";
import { headerModalProps } from "../../internal/helpers";

const platformName = "IBM Platform";
const sendIdeasUrl = "https://ideas.ibm.com";

describe("Feedback", () => {
  test("snapshot", async () => {
    const { baseElement } = render(
      <Feedback platformName={platformName} sendIdeasUrl={sendIdeasUrl} {...headerModalProps} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  test("functional", async () => {
    render(<Feedback platformName={platformName} sendIdeasUrl={sendIdeasUrl} {...headerModalProps} />);

    expect(await screen.findByText("We look forward to your feedback!")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button", { name: /OK/i }));
  });

  test("a11y", async () => {
    const { container } = render(
      <Feedback platformName={platformName} sendIdeasUrl={sendIdeasUrl} {...headerModalProps} />
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
