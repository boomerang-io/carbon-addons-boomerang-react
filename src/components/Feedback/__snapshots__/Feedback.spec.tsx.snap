// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Feedback > snapshot 1`] = `
<body>
  <div>
    <div
      aria-hidden="false"
      class="cds--layer-one cds--modal is-visible cds--bmrg-feedback-container cds--bmrg-header-modal"
      role="presentation"
    >
      <div
        aria-label="Feedback"
        aria-modal="true"
        class="cds--modal-container"
        role="dialog"
      >
        <button
          class="cds--visually-hidden"
          type="button"
        >
          Focus sentinel
        </button>
        <div
          class="cds--modal-container-body"
        >
          
          <div
            class="cds--modal-header"
          >
            <h2
              class="cds--modal-header__heading cds--type-beta"
            >
              Submit an Idea
            </h2>
            <div
              class="cds--modal-close-button"
            >
              <span
                class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--left cds--tooltip cds--icon-tooltip"
              >
                <div
                  class="cds--tooltip-trigger__wrapper"
                >
                  <button
                    aria-label="Close"
                    aria-labelledby="tooltip-2"
                    class="cds--btn--icon-only cds--modal-close cds--btn cds--btn--primary"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="cds--modal-close__icon"
                      fill="currentColor"
                      focusable="false"
                      height="20"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 32 32"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.4141 16L24 9.4141 22.5859 8 16 14.5859 9.4143 8 8 9.4141 14.5859 16 8 22.5859 9.4143 24 16 17.4141 22.5859 24 24 22.5859 17.4141 16z"
                      />
                    </svg>
                  </button>
                </div>
                <span
                  aria-hidden="true"
                  class="cds--popover"
                  id="tooltip-2"
                  role="tooltip"
                >
                  <span
                    class="cds--popover-content cds--tooltip-content"
                  >
                    Close
                  </span>
                  <span
                    class="cds--popover-caret"
                  />
                </span>
              </span>
            </div>
          </div>
          <div
            class="cds--layer-two cds--modal-content"
          >
            <div
              class="cds--bmrg-feedback"
            >
              <p>
                Share your thoughts and ideas on what we can do to improve the 
                IBM Platform
                 platform and our onboarding process.
              </p>
              <p>
                To submit your idea, visit our portal at
                 
                <a
                  aria-describedby="new-window-aria-desc-0"
                  href="https://ideas.ibm.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  https://ideas.ibm.com
                </a>
                .
              </p>
              <p>
                For now, you must have an 
                 email address and you will need to register with a password during your first visit.
              </p>
              <p>
                You will have the opportunity to see other public ideas, vote on them and track the status of your idea.
              </p>
              <p>
                We look forward to your feedback!
              </p>
            </div>
          </div>
          <div
            aria-busy="false"
            class="cds--modal-footer cds--btn-set"
          >
            <button
              class="cds--btn cds--btn--primary"
              data-modal-primary-focus="true"
              type="button"
            >
              OK
            </button>
          </div>
        </div>
        <button
          class="cds--visually-hidden"
          type="button"
        >
          Focus sentinel
        </button>
      </div>
    </div>
  </div>
</body>
`;
